src/compiler/checker.ts

```diff
@@ -34310,87 +34313,95 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
 
         function chooseOverload(candidates: Signature[], relation: Map<string, RelationComparisonResult>, isSingleNonGenericCandidate: boolean, signatureHelpTrailingComma = false) {
-            candidatesForArgumentError = undefined;
-            candidateForArgumentArityError = undefined;
-            candidateForTypeArgumentError = undefined;
-
-            if (isSingleNonGenericCandidate) {
-                const candidate = candidates[0];
-                if (some(typeArguments) || !hasCorrectArity(node, args, candidate, signatureHelpTrailingComma)) {
-                    return undefined;
-                }
-                if (getSignatureApplicabilityError(node, args, candidate, relation, CheckMode.Normal, /*reportErrors*/ false, /*containingMessageChain*/ undefined)) {
-                    candidatesForArgumentError = [candidate];
-                    return undefined;
+            chooseOverloadRecursionLevel++; // #56013
+            chooseOverloadFlushNodesSignaturesReq[chooseOverloadRecursionLevel] = undefined;
+            const result = (() => {

```

Comment:
Just a style thing - extract the IIFE into a named function in the scope, such as `chooseOverloadWorker` instead; makes debugging easier later for it to be named.